src/components/pages/gallery/Upload.tsx

```diff
@@ -245,10 +302,17 @@ export default function Upload(props: Props) {
             props.setUploadInProgress(true);
             props.closeCollectionSelector();
             await props.syncWithRemote(true, true);
+            // set state to not done
+            await ImportService.setToUploadFiles(
```

Comment:
so we are just setting and ToUploadFiles once at the start, and so if the upload fails midway, the files which were successfully uploaded would still be retired no? 

ideally, we should update this list after each successful upload.if this is not too difficult we should do this